### PR TITLE
Change path for local blockchain state

### LINKED ISSUE #189: MESG Framework
Improve the CLI for local development
- [x] Simplify the management on the local engine (https://github.com/mesg-foundation/js-sdk/pull/188)
  - [x] start and stop engine automatically in `dev` commands (https://github.com/mesg-foundation/js-sdk/pull/188)
  - [x] inject account in the engine (https://github.com/mesg-foundation/js-sdk/pull/188)
- [x] Remove confusing commands (list, detail, create, compile, delete, daemon) (https://github.com/mesg-foundation/js-sdk/pull/188)
- [x] MESG framework (https://github.com/mesg-foundation/js-sdk/pull/198)
- [ ] update docs.mesg.com once done (https://github.com/mesg-foundation/docs/pull/121)

### PR DESCRIPTION
**Pull request by antho1404**
dependency: https://github.com/mesg-foundation/js-sdk/pull/196

The local state (`.mesg` directory) is now located in the dataDir of the user and not the current path.

https://oclif.io/docs/config
> dataDir - CLI data directory
    Unix: ~/.data/mycli
  Windows: %LOCALAPPDATA%\mycli
  Can be overridden with XDG_DATA_HOME

I also removed the flag in order to simplify the cli and the `configFile` was not read anyway by the engine.

### CODE REVIEW

**Review by NicolasMahe — COMMENTED** [review]
I got the error:
```
    ✖ Creating default configuration
      → ENOENT: no such file or directory, mkdir '/Users/nico/.local/share/@mesg/cli'
```
fixed by running `mkdir -p /Users/nico/.local/share/@mesg/cli`

**Comment by NicolasMahe** [comment]
I now have weird behavior when even if the command `service:dev` works without issue, it doesn't stop the engine and remove the `.mesg`. The `pid.json` file still exists as well as the engine's config and files. The `pid.json` file contains:
```json
[86371,86380]
```

**Comment by NicolasMahe** [comment]
When the command `service:dev` throws an error, the engine is not stopped and the next use of `service:dev` doesn't stop the engine and the running service:
```
➜  js-sdk git:(bug/config-path) ./packages/cli/bin/run service:dev ../services/service-ethereum
  ✔ Start environment
  ✔ Compiling service
  ✖ Creating service
    → out of gas: out of gas in location: WritePerByte; gasWanted: 300000, gasUsed: 347768
    Starting service
    Fetching service's logs
Error: out of gas: out of gas in location: WritePerByte; gasWanted: 300000, gasUsed: 347768
    at API.<anonymous> (~/Development/MESG/js-sdk/packages/api/lib/lcd.js:58:23)
    at Generator.next (<anonymous>)
    at fulfilled (~/Development/MESG/js-sdk/packages/api/lib/lcd.js:5:58)



➜  js-sdk git:(bug/config-path) ./packages/cli/bin/run service:dev ../services/service-emit-event-interval
  ✔ Start environment
  ✔ Compiling service
  ✔ Creating service
  ✔ Starting service
  ✔ Fetching service's logs
(node:1) UnhandledPromiseRejectionWarning: Error: 2 UNKNOWN: create event: data invalid request: instance "En6ewB6q5Gh5s7V7E4ftTzkbFSV2ozeHgTB8QN7wMWAv" not found
    at Object.exports.createStatusError (/app/node_modules/grpc/src/common.js:91:15)
    at Object.onReceiveStatus (/app/node_modules/grpc/src/client_interceptors.js:1209:28)
    at InterceptingListener._callNext (/app/node_modules/grpc/src/client_interceptors.js:568:42)
    at InterceptingListener.onReceiveStatus (/app/node_modules/grpc/src/client_interceptors.js:618:8)
    at callback (/app/node_modules/grpc/src/client_interceptors.js:847:24)
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
EVENT[every_2_seconds]: {"timestamp":1584706450623}
EVENT[every_1_second]: {"timestamp":1584706450638}
EVENT[every_1_second]: {"timestamp":1584706451643}
EVENT[every_2_seconds]: {"timestamp":1584706452623}
EVENT[every_1_second]: {"timestamp":1584706452646}
EVENT[every_5_seconds]: {"timestamp":1584706453623}
EVENT[every_1_second]: {"timestamp":1584706453648}
EVENT[every_2_seconds]: {"timestamp":1584706454591}
EVENT[every_1_second]: {"timestamp":1584706454615}
^C  ✔ Stopping logs
  ✔ Stop environment
 ›   Warning: Event stream error: 1 CANCELLED: Cancelled
 ›   Warning: Result stream error: 1 CANCELLED: Cancelled


➜  js-sdk git:(bug/config-path) docker service list
]ID                  NAME                                                  MODE                REPLICAS            IMAGE                                                                     PORTS
fmp8nhmaj1co        engine                                                replicated          1/1                 mesg/engine                                                               *:1317->1317/tcp, *:26656->26656/tcp, *:50052->50052/tcp
iitonbjnjx28        engine-8nMhm14WhFSZr2dGAgrH1xe86mUZLrQUNY2jnbBzoQ3G   replicated          1/1                 sha256:d84c5f68c5994d42bdc2aab41a88e8a457f86380b26fd051f90adc321ad115b4
```

File: packages/cli/CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Commands `service:dev` and `process:dev` are automatically start a local environ
 - [#193](https://github.com/mesg-foundation/js-sdk/pull/193) Add login/logout commands with synchronization in the user's configs
 - [#194](https://github.com/mesg-foundation/js-sdk/pull/194) Add `deploy:service` command
 - [#196](https://github.com/mesg-foundation/js-sdk/pull/196) Update code organization for tasks
+- [#](https://github.com/mesg-foundation/js-sdk/pull/) State of the local chain in the `dataDir` (https://oclif.io/docs/config)
```
Comment (antho1404) [inline packages/cli/CHANGELOG.md]:
```suggestion
- [#197](https://github.com/mesg-foundation/js-sdk/pull/197) State of the local chain in the `dataDir` (https://oclif.io/docs/config)
```

**Comment by antho1404** [comment]
first issue fixed and second in a new issue https://github.com/mesg-foundation/js-sdk/issues/201

**Review by NicolasMahe — APPROVED** [review]

**Pull request merged**